dlt/load/utils.py

```diff
@@ -251,19 +249,30 @@ def filter_new_jobs(
     if mp := capabilities.max_parallel_load_jobs:
         max_workers = min(max_workers, mp)
 
+    # if all slots are full, do not create new jobs
+    if len(running_jobs) >= max_workers:
```

Comment:
this must be checked before you call this function. do not pass `running_jobs` here. you are mixing two separate things

Reply:
done